daemon/logger/splunk/splunk.go

```diff
@@ -402,6 +409,14 @@ func (l *splunkLogger) queueMessageAsync(message *splunkMessage) error {
 	if l.closedCond != nil {
 		return fmt.Errorf("%s: driver is closed", driverName)
 	}
+	if l.logNonBlocking {
+		select {
+		case l.stream <- message:
+			return nil
+		default:
+			return fmt.Errorf("%s: buffer is full", driverName)
```

Comment:
Discussing with some maintainers; we should probably avoid situations where this will result in a large volume of logs being sent to the daemon logs. Perhaps we need to log this every XX message ("100 logs entries failed to be logged"), or something along those lines.

Otherwise a connection issue could result in the daemon logs being spammed.

Reply:
Indeed. There is rate limiting on the number of error messages logged in place already, set to log a maximum of 10000 logs per 30 seconds, if I recall correctly. See logger_error.go.